build_native

```diff
@@ -8,6 +8,10 @@ cd build/$HOSTOS/$ARCH
 if [ -e "$TOOLCHAIN_FILE" ]; then
     ARGS=-DCMAKE_TOOLCHAIN_FILE=$TOOLCHAIN_FILE
 fi
-cmake --config Release -DENABLE_BZIP2=OFF -DENABLE_LZMA=OFF -DCMAKE_C_FLAGS="-O2" -GNinja -DCMAKE_MAKE_PROGRAM=ninja -DBUILD_SHARED_LIBS=ON -DOPENSSL_USE_STATIC_LIBS=ON -DENABLE_BZIP2=OFF -DENABLE_LZMA=OFF $ARGS ../../../external/libzip
+if [ "$HOSTOS" == "Darwin" ]; then
+    OSXARGS=-DCMAKE_OSX_ARCHITECTURES="x86_64;arm64"
```

Comment:
Perhaps instead of specifying the architectures here, it would be better to add [this code](https://github.com/mono/mono.posix/blob/b35857a998c81529b441260171ad1262b9d014a3/src/native/CMakeLists.txt#L496-L541) to CMakeLists.txt?

Reply:
In this case the `CMakeLists.txt` file is being executed from the `external/libzip` directory. So we would either need to submit a PR upstream and then wait for a release or apply another patch if we wanted to make all of those changes. 

I figured just passing the additional arguments to the `cmake` command was the simplest solution since it works and it produces the universal binary we need.